### DISCUSSION TITLE
Question: is possible to build a fully custom application to run it on Onion?

### DISCUSSION THREAD

**Discussion by Wohlstand** [OnionUI/Onion#1269] [Q&A]
Hello!

I am developer of a FOSS game that has support of many platforms to be run (including various homebrew targets). The game uses SDL2 for everything, but in some cases, there are custom modules being used to provide certain OS API parts.

And the main question: does OnionOS allows to build a fully custom application to run it on the OnionOS?

I am not familiar with OnionOS by myself, but I got been asked by a user of my game about this case.

EDIT: While brosed a documentations, I found something related to "ports": https://onionui.github.io/docs/ports#how-to-add-your-own-ports-for-advanced-users

But, I am not sure, what cross toolchain I need to, and what API the OS do support? I work on Linux by myself (on hardware), so, I have totally no trobules with cross toolchains, they works just easy.

**Comment by Wohlstand** [comment]
Ping?

**Comment by XK9274** [comment]
Hey! 

Here's some repos that incorporate an SDL2 frontend and run on MM/MMP:

- https://github.com/Yorisoft/pokedex_miyoo 
- https://github.com/XK9274/moonlight-embedded-miyoo/blob/master/src/sdl.c
- https://github.com/LostQuasar/SDL2_MiyooMiniExample

Some have app structures to get you started once you're ready to run it on the miyoo

A lot of us use this toolchain: https://github.com/shauninman/union-miyoomini-toolchain for cross compiling, you'll then need to package it into an App directory, which you can see on the repo i've shared by Yorisoft. 

Here's some custom SDL2 libs by Steward (Who built the majority of the new mmiyoo backend) and some modified by me:
- https://github.com/steward-fu/nds
- https://github.com/XK9274/sdl2_miyoo/branches

You'll find some are directed at particular binaries, there's a vanilla branch on my repo that i find works with a lot of things however it's not been rebased in a while so performance is just "ok"

Some binaries work out of the box with these SDL2 libs, others don't.

**Reply by Wohlstand** [reply]
Thanks for the reply! I'll try to test out some. I will also need to ask somebody to try out the build on the hardware...
